pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go

```diff
@@ -99,14 +101,21 @@ func (a *Authenticator) AuthenticatePassword(username, password string) (user.In
 		return nil, false, err
 	}
 
-	if len(remoteUserData.ID) == 0 {
+	if len(remoteUserData.Subject) == 0 {
 		return nil, false, errors.New("Could not retrieve user data")
 	}
+	identity := authapi.NewDefaultUserIdentityInfo(a.providerName, remoteUserData.Subject)
+
+	if len(remoteUserData.Name) > 0 {
```

Comment:
Not that I dislike cleanliness of data, but why bother?  We're already checking at the other end and I'm hard pressed to explain a significant distinction between present and empty and not-present in this case.